Cortex Attack Surface Management 2.0.0

johnnywilkes opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Consider this a re-imaging of the "Cortex ASM - ASM Alert" playbook to make it useful to handle ASM related alerts.  Please let us know if we should submit a demo video or if you would like to do it live!

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [X] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

johnnywilkes commented:
@ShacharKidor , are you able to take look at this, please?

ShacharKidor commented:
> @ShacharKidor / @idovandijk / @altmannyarden are you able to take look at this, please?

Hi @johnnywilkes - a reviewer from our contribution team will start the review soon. 

@dorschw - FYI.

idovandijk reviewed (commented):
Thank you for yet another contribution @johnnywilkes, your time and effort are greatly appreciated.
Here are my comments for the playbooks:

### Cortex ASM - Detect Service
1. Should this be OR or AND? The operator currently applied is "and" but the title says "or". Also later on in the Cortex ASM - ASM Alert playbook, there is a task that sets the value of "None" if there is no remote IP, which suggests to me that there is not always an IP in the alert:
![image](https://user-images.githubusercontent.com/43602124/199717902-72370d31-9773-4aad-956c-3cf5ffb57407.png)
![image](https://user-images.githubusercontent.com/43602124/199718394-8ae6999b-126b-4bf1-a0e2-f4bd87226174.png)


### Cortex ASM - ASM Alert
2. Question - is there a scenario where this grid field gets more than one entry? If so, that can interfere with this condition:
![image](https://user-images.githubusercontent.com/43602124/199715829-c61178cd-b338-4ac0-9cfa-7276a5681e5c.png)
If the answer to the question is yes, you may want to consider one of the following changes:
A. Using a playbook output to determine if the port is closed, rather than checking the grid field values.
B. Changing the condition to "GET `alert.asmservicedetection` WHERE `alert.asmservicedetection.ScanResult` DOESN'T EQUAL `Closed` - IS DEFINED" - which would check if any entry has a value other than "closed".
If there will always be one entry there, then no change is required here.
3. What data does the "ASM Tags" field hold? Why does the occurrence of "dev" imply that automated remediation can be done? E.g., is this something that should be configurable?
![image](https://user-images.githubusercontent.com/43602124/199721054-3c30a56d-f6a3-4f8e-a067-77482f78a06e.png)

Please let me know when we can do a short demo for this. Feel free to reach out if you have any questions.

troberts13 commented on Packs/CortexAttackSurfaceManagement/Scripts/GenerateASMReport/GenerateASMReport_test.py:
```diff
@@ -0,0 +1,72 @@
+
+
+def test_get_asm_args(mocker):
+    from GenerateASMReport import get_asm_args
```
@dorschw  where is this documented for reference? I don't see it mentioned here: https://xsoar.pan.dev/docs/integrations/unit-testing

johnnywilkes replied:
@dorschw , we have added these, let us know what you think and if we should add some type of documentation to: https://xsoar.pan.dev/docs/integrations/code-conventions#unit-tests

dorschw replied:
thanks for the suggestion, it should be fixed in [this PR](https://github.com/demisto/content-docs/pull/1223)

johnnywilkes commented on Packs/CortexAttackSurfaceManagement/Scripts/GetRemediationGuidance/GetRemediationGuidance.py:
we are replacing that automation withe a playbook/list (Cortex ASM - Remediation Guidance/RemediationGuidance) so I don't need to make a change here

johnnywilkes commented on Packs/CortexAttackSurfaceManagement/Scripts/GenerateASMReport/GenerateASMReport.yml:
```diff
@@ -0,0 +1,56 @@
+  name: alert_severityStr
+- description: Potential service owners gathered through playbook.
+  isArray: true
+  name: asmserviceowner
```
@dorschw , changes we made:
asmserviceowner -> asm_service_owner
asmremediation -> asm_remediation
asmservicedetection -> asm_service_detection
asmsystemids -> asm_system_ids
asmcloud -> asm_cloud
asmnotification -> asm_notification
asmdatacollection -> asm_data_collection
asmtags -> asm_tags
asmprivateip -> asm_private_ip
asmrelated -> asm_related

johnnywilkes commented:
@idovandijk / @dorschw 
We have made all changes we talked about Thursday. In Summary:

Ido's #1 comment:
- In "Cortex ASM - Detect Service" we changed the first conditional to "and" instead of "or" (title and description also)
<img width="830" alt="image" src="https://user-images.githubusercontent.com/32227961/200041748-0a730ee2-4829-4b6a-ac49-ea93851e8965.png">
- In "Cortex ASM - Enrichment" we made RemoteIP input no longer mandatory and simplified the first conditional of the playbook (removed 2nd expression):
<img width="483" alt="image" src="https://user-images.githubusercontent.com/32227961/200042458-6db714a1-374d-45ca-bed4-da32b3592cff.png">
- In "Cortex ASM - ASM Alert" we removed the "SetIfEmpty" transform for the "Cortex ASM - Enrichment" subplaybook as it was no longer needed:
<img width="490" alt="image" src="https://user-images.githubusercontent.com/32227961/200042500-b9996cef-923f-44d9-b137-b36ec561c12e.png">

Ido's #2 comment: We verified with our engineering group that ${alert.RemoteIP} and ${alert.RemotePort} should only be at most a single entry.

Per @dorschw 's suggestion, we removed the "GetRemediationGuidance" automation and replaced it with a new subplaybook (Cortex ASM - Remediation Guidance) and list (RemediationGuidance).  @idovandijk , i am assuming you will have to look over that playbook as well.

Also changed inputs and adding unit-test docstrings to GenerateASMReport automation.


Please let me know if you have any comments, questions or concerns.  Thank you again for your assistance.

dorschw commented on Packs/CortexAttackSurfaceManagement/ReleaseNotes/2_0_0.md:
```diff
@@ -0,0 +1,35 @@
+#### Playbooks
+
+##### Cortex ASM - ASM Alert
+Replaced earlier **Cortex ASM - ASM Alert** playbook and Readme files.
```
what is changed in the playbook? What difference should users expect?

johnnywilkes replied:
@dorschw , the original playbook was created really just as an "example" and not sure it was actually used.  It focused on enrichment whereas now there are additional stages (triage, enrichment, decision, remediation, ticketing, validation, summarization).  Let me know if you want me to add more data in this release note.

dorschw replied:
Please mention the additional stages (`The playbook now allows ...`)

dorschw commented on Packs/CortexAttackSurfaceManagement/pack_metadata.json:
```diff
@@ -2,7 +2,7 @@
     "description": "Content for working with Attack Surface Management (ASM).",
     "support": "xsoar",
-    "currentVersion": "1.1.0",
+    "currentVersion": "2.0.0",
```
Major version bumps usually mark breaking changes, or significant changes - is this the case here? If it's a feature and backwards compatible, I'd change it to a minor bump (1.1.0 -> 1.2.0)

johnnywilkes replied:
@dorschw , as mentioned in the previous comment there were some significant changes in the main playbook.  I don't really consider them "breaking" as the previous playbook was just setup as and example, and don't believe it was being used.  That being said, I don't mind using any version number that you recommend.  Let me know what change (if any) I should make, please.

dorschw replied:
let's do `1.2.0`, just change the release note file name, and the version value under `pack_metadata.json`, and we'll be ready to merge it :)

johnnywilkes replied:
@dorschw , just pushed changes

dorschw reviewed (commented):
Two minor comments, aside from this, the PR is approved on my end.

ShirleyDenkberg commented:
@idovandijk @troberts13 @dorschw Doc review completed.

idovandijk reviewed (commented):
Reviewed the recent playbook changes, looks great!
@dorschw feel free to approve this PR once you're done with the code review.

dorschw reviewed (approved):
🎉
